org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/info/InfoWindow.java

```diff
@@ -624,6 +626,40 @@ protected boolean loadInfoDefinition() {
 				vo.IsReadOnly = infoColumn.isReadOnly();
 				GridField gridField = new GridField(vo);
 				gridFields.add(gridField);
+
+				//IDEMPIERE-4485 Store new Gridfields with IsReadOnly = false
+				if(infoColumn.isQueryCriteria()) {
+					vo = GridFieldVO.createParameter(infoContext, p_WindowNo, AEnv.getADWindowID(p_WindowNo), infoWindow.getAD_InfoWindow_ID(), 0,
+							columnName, infoColumn.getNameTrl(), infoColumn.getAD_Reference_ID(), 
+							infoColumn.getAD_Reference_Value_ID(), isMandatory, false, infoColumn.getPlaceHolderTrl());
+					
+					if (infoColumn.getAD_Val_Rule_ID() > 0) {
+						vo.ValidationCode = infoColumn.getValidationCode();
+						if (vo.lookupInfo != null) {
+							vo.lookupInfo.ValidationCode = vo.ValidationCode;
+							vo.lookupInfo.IsValidated = false;
+						}
+					}
+					if (infoColumn.getDisplayLogic() != null)					
+						vo.DisplayLogic =  infoColumn.getDisplayLogic();
+					if (infoColumn.isQueryCriteria() && infoColumn.getDefaultValue() != null)
+						vo.DefaultValue = infoColumn.getDefaultValue();
+					desc = infoColumn.getDescriptionTrl();
+					vo.Description = desc != null ? desc : "";
+					help = infoColumn.getHelpTrl();
+					vo.Help = help != null ? help : "";
+					vo.AD_FieldStyle_ID = infoColumn.getAD_FieldStyle_ID();
+					vo.IsAutocomplete = infoColumn.isAutocomplete();
```

Comment:
the code before vo.IsReadOnly can be shorten to the following:
vo = vo.clone(infoContext, p_WindowNo, 0, vo.AD_Window_ID, 0, false);

Reply:
Commited changes